ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -139,6 +154,29 @@ public void run() {
   }
 
   /**
+   * Pauses hard deletes. If hard deletes is half way in hard deleting a range, it will continue to finish hard deleting
+   * the range and then pause. Hard delete will be in paused state until {@link #resume()} is called.
+   */
+  void pause() throws InterruptedException {
+    reentrantLock.lock();
+    if (!paused.getAndSet(true)) {
+      logger.info("HardDelete thread has been paused ");
+    }
+    reentrantLock.unlock();
```

Comment:
Three points of confusion I have here,

1. One of the requirements of pause that we have outlined from the beginning is that the `pause()` should reliably pause hard deletes i.e. when the `pause()` call returns, hard delete should be paused.
2. Where is the persisting of index here that guarantees that guarantees that the log is flushed and hard delete will have no recovery to do?
3. Where is the hard delete recovery range updated to indicate that there will be recovery to do?